### PR TITLE
Implement JavaLibrary and add initial test framework.

### PR DESCRIPTION
**Pull request by azabbasi**

### CODE REVIEW

File: sdk/modelparser/azure-digitaltwins-parser/src/test/java/com/azure/digitaltwins/parser/ClassCodeGeneratorTests.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+import java.io.IOException;
+
+/**
+ * Tests code generation for {@link JavaClass}.
+ * Note that The tests make the following assumptions:
+ * - Each test has a field called: "typeName" which will dictate the name of each class.
+ * - Each test expects an existing file with the "typeName" and ".expected" extension in the src/test/resources/ClassTestResources
+ * - Each test will have a code comment with the path to the target files.
+ * - Each test generates a file with ".temp.generated" extension. generated files will be deleted after the test pass.
```
Comment (barustum) [inline sdk/modelparser/azure-digitaltwins-parser/src/test/java/com/azure/digitaltwins/parser/ClassCodeGeneratorTests.java:17]: upper case G in 'generated files will be delete...'

File: sdk/modelparser/azure-digitaltwins-parser/src/test/java/com/azure/digitaltwins/parser/ClassCodeGeneratorTests.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+        javaClass.generateCode(codeWriter);
+
+        codeWriter.close();
+
+        this.compareGeneratedCodeWithExpected(TEST_SUB_DIRECTORY, typeName);
+    }
+
+    /**
+     * Tests an empty package private java class with no class comments.
```
Comment (barustum) [inline sdk/modelparser/azure-digitaltwins-parser/src/test/java/com/azure/digitaltwins/parser/ClassCodeGeneratorTests.java:81]: an empty public abstract class

File: sdk/modelparser/azure-digitaltwins-parser/src/test/java/com/azure/digitaltwins/parser/FileHelpers.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+package com.azure.digitaltwins.parser;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class FileHelpers {
```
Comment (barustum) [inline sdk/modelparser/azure-digitaltwins-parser/src/test/java/com/azure/digitaltwins/parser/FileHelpers.java:12]: public class comments.

File: sdk/modelparser/azure-digitaltwins-parser/src/test/java/com/azure/digitaltwins/parser/GeneratedCodeCompareBase.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+// Licensed under the MIT License.
+
+package com.azure.digitaltwins.parser;
+
+import com.azure.digitaltwins.parser.implementation.codegen.CodeWriter;
+import org.junit.jupiter.api.Assertions;
+
+import java.io.IOException;
+
+public class GeneratedCodeCompareBase {
```
Comment (barustum) [inline sdk/modelparser/azure-digitaltwins-parser/src/test/java/com/azure/digitaltwins/parser/GeneratedCodeCompareBase.java:11]: public class comments

File: sdk/modelparser/azure-digitaltwins-parser/src/test/java/com/azure/digitaltwins/parser/FileHelpers.java (hunk 3 above)
Comment (azabbasi) [reply sdk/modelparser/azure-digitaltwins-parser/src/test/java/com/azure/digitaltwins/parser/FileHelpers.java:12]: made it package internal

File: sdk/modelparser/azure-digitaltwins-parser/src/test/java/com/azure/digitaltwins/parser/GeneratedCodeCompareBase.java (hunk 4 above)
Comment (azabbasi) [reply sdk/modelparser/azure-digitaltwins-parser/src/test/java/com/azure/digitaltwins/parser/GeneratedCodeCompareBase.java:11]: made class internal.

**Review by barustum — APPROVED** [review]

**Pull request merged**